src/derived.rs

```diff
@@ -44,7 +48,8 @@ where
     DB: Database,
     MP: MemoizationPolicy<DB, Q>,
 {
-    map: RwLock<FxHashMap<Q::Key, QueryState<DB, Q>>>,
+    lru_cap: AtomicUsize,
```

Comment:
We deliberately store `lru_cap` outside of `QueryMap` (although it logically belongs there!) such that we can read it wihtout locking

Reply:
can you put a comment about why the lru-cap is stores outside of map?